### PR TITLE
Various Bug Fixes and unnecessary code removal episode 2 (actually works edition)

### PR DESCRIPTION
**Pull request by Fear-MK**
## Description
Fixed following open issues:
- Implemented "Squad Queue" tier from recent Lounge API update (Untested)
- Fixed Predict command and eradicated all URL shortening code (useless as tinyurl API has become unavailable) (Untested)
- Fixed list formatting issues within certain commands (Tested)
## Checklist

- [ ] If code changes were made, they have been tested
- [x] This PR fixes an issue
- [ ] This PR is not a code change (ie. documentation, typehinting, comments, etc.)

### CODE REVIEW

File: BadWolfBot.py [hunk 1]
```diff
@@ -1142,9 +1138,6 @@ def read_next_token(file_handle, seperation_key = ":") -> str:
         real_bot_key = read_next_token(f)
         beta_bot_key = read_next_token(f)
         testing_bot_key = read_next_token(f)
-        URLShortener.BITLY_API_TOKEN = read_next_token(f)
```
Comment (BadWolf1023) [inline BadWolfBot.py:1145]: Please update `private_example.txt` to comply with these new changes.

File: MogiUpdate.py [hunk 2]
```diff
@@ -170,7 +130,7 @@ def createJSON(players, mult=default_multiplier):
 # - Individual player races played will be the actual number of races they played as well
 # - Multiplier for each person will be global races played ÷ 12
 # - Updater Bot will leave all multipliers alone, even on subs/subbees. Updater Bot will not change any JSON except for the following: Updater Bot must change gain/loss prevention and full gain/loss on JSON appropriately for sub ins and sub outs.
-def create_player_json(player:Tuple[str, int, int, int], races_played=12, sub_in=False, sub_out=False, squadqueue=False):
```
Comment (BadWolf1023) [inline MogiUpdate.py:173]: Nice catch

File: MogiUpdate.py
Comment (BadWolf1023) [inline MogiUpdate.py:1]: Squad queue update looks like good work

File: TagAIShell.py [hunk 3]
```diff
@@ -169,7 +169,7 @@ def nice_print_teams(teams, num_tabs=0):
         result += '\t'*num_tabs + f"Tag: {teamTag}\n"
         for fc, playerName in fc_players:
             playerNum += 1
-            result += '\t'*num_tabs + f"\t{playerNum}. {playerName}\n"
+            result += '\t'*num_tabs + f"\t{playerNum}\. {playerName}\n"
```
Comment (BadWolf1023) [inline TagAIShell.py:172]: No, this isn't Discord text

File: Room.py [hunk 4]
```diff
@@ -629,7 +629,7 @@ def get_sorted_player_list_string(self, startrace=None, endrace=None, lounge_rep
         to_build = ""
         for counter, (fc, player) in enumerate(players, 1):
             fc_str = f" - {fc}" if include_fc else ""
-            to_build += f"{counter}. {UserDataProcessing.proccessed_lounge_add(player, fc, lounge_replace)}{fc_str}\n"
+            to_build += f"\t{counter}\. {UserDataProcessing.proccessed_lounge_add(player, fc, lounge_replace)}{fc_str}"
```
Comment (BadWolf1023) [inline Room.py:632]: This looks buggy to me - we removed the new line and added a tab instead...? Please show screenshots of what this change looks like.

File: Race.py [hunk 5]
```diff
@@ -444,7 +444,7 @@ def __str__(self):
          "\nMatch end time: " + str(self.matchTime)
         placementsSTR = ""
         for placement in self.placements:
-            placementsSTR += str("\n\t" + str(placement))
+            placementsSTR += str("\n" + str(placement))
```
Comment (BadWolf1023) [inline Race.py:447]: This looks like an unintended change to our display. Please show screenshots of what this change looks like.

File: commands.py [hunk 6]
```diff
@@ -1326,15 +1325,6 @@ async def _mogi_update(client, message: discord.Message, this_bot: TableBot.Chan
...
 
-                shortened_admin_panel_link = "No Link"
-                try:
-                    admin_link_tiny_url = await URLShortener.tinyurl_shorten_url(updater_link)
-                    shortened_admin_panel_link = f"[Preview]({admin_link_tiny_url})"
-                except:
-                    pass
-
-                embed.add_field(name='Short Preview Link:', value=shortened_admin_panel_link)
-
```
Comment (BadWolf1023) [inline commands.py:1337]: Now they can't preview the update at all. Please see #186 for exact instructions.

File: commands.py [hunk 7]
```diff
@@ -1355,19 +1345,6 @@ async def _mogi_update(client, message: discord.Message, this_bot: TableBot.Chan
...
-                    if updater_link == preview_link:
-                        shortened_preview_link = shortened_admin_panel_link
-                    else:
-                        preview_link_tiny_url = await URLShortener.tinyurl_shorten_url(preview_link)
-                        shortened_preview_link = f"[Preview]({preview_link_tiny_url})"
-                except:
-                    pass
-
-                embed.add_field(name='Short Preview Link:', value=shortened_preview_link)
-
```
Comment (BadWolf1023) [inline commands.py:1370]: Now they can't preview the update at all. Please see https://github.com/BadWolf1023/MKW-Table-Bot/issues/186 for exact instructions.

File: logging/feedback_logs.txt [hunk 8]
```diff
@@ -1 +1 @@
-2021-06-27 15:49:45.099623: Bad Wolf#1024 - 706120725882470460:log I really like the mii heads!
\ No newline at end of file
+2021-06-27 15:49:45.099623: Bad Wolf#1024 - 706120725882470460:log I really like the mii heads! shut up
```
Comment (BadWolf1023) [inline logging/feedback_logs.txt:1]: Let's keep the main repo professional (so remove).

**Review by BadWolf1023 — CHANGES_REQUESTED** [review]
Please see my comments.

File: TagAIShell.py (hunk 3 above)
Comment (Fear-MK) [reply TagAIShell.py:172]: Sorry must have been a bi-product of testing all sorts of random stuff and forgetting to remove

File: Room.py (hunk 4 above)
Comment (Fear-MK) [reply Room.py:632]: yes it does not sure what I did here, going to run this code tmw, and see what it looked like

File: Race.py (hunk 5 above)
Comment (Fear-MK) [reply Race.py:447]: Was sent in DMs previously. It meant an ornate list, but different formatting. I commented but you said it was fine (will find original message later

File: commands.py (hunk 7 above)
Comment (Fear-MK) [reply commands.py:1370]: Yep my mistake, I assumed there were two link boxes. Didnt check specific instructions

File: Race.py (hunk 5 above)
Comment (Fear-MK) [reply Race.py:447]: image

File: logging/feedback_logs.txt (hunk 8 above)
Comment (Fear-MK) [reply logging/feedback_logs.txt:1]: fine

File: BadWolfBot.py (hunk 1 above)
Comment (Fear-MK) [reply BadWolfBot.py:1145]: Will try if I can sync branch after next commit

File: Race.py (hunk 5 above)
Comment (Fear-MK) [reply Race.py:447]: Will wait on confirmation that this is satisfactory

Comment (camelwater) [reply Race.py:447]: I like the tab. It spaces out the placements from the race details

Comment (Fear-MK) [reply Race.py:447]: > I like the tab. It spaces out the placements from the race details

the tab is what was causing peculiar format issues

Comment (Fear-MK) [reply Race.py:447]: Gonna quickly try something, actually

Comment (Fear-MK) [reply Race.py:447]: Command has been reverted to its original form.

File: commands.py (hunk 7 above)
Comment (Fear-MK) [reply commands.py:1370]: Renamed Short Preview Link to Preview Link and supplied full link to field

Comment (Fear-MK) [reply commands.py:1370]: I lied, this isnt complete, iirc

File: Room.py [hunk 9]
```diff
@@ -71,6 +71,7 @@ def __init__(self, table: 'ChannelBot', rxx: str, races: List[Race.Race], event_
         self.set_up_user_display_name = setup_display_name
         self.sub_ins: Dict[str, Dict[str, Any]] = {} #dict of FCs that subbed in: dict(sub_in_start_race, sub_in_end_race, sub_out_fc, sub_out_mii_name, sub_out_name, sub_out_start_race, sub_out_end_race, sub_out_scores)
         self.is_freed = False
+        self.multiple_rxxs_used = False
```
Comment (BadWolf1023) [inline Room.py:74]: Not necessary and prone to bugs if `/undo` is run. Make a new function:
```
def has_multiple_rxxs(self):
   return len(self.rLIDs) > 1
```

File: commands.py [hunk 10]
```diff
@@ -1293,7 +1292,7 @@ async def _mogi_update(client, message: discord.Message, this_bot: TableBot.Chan
 
                 if using_table_bot_table:
                     war_had_errors = len(this_bot.getWar().get_all_war_errors_players(this_bot.getRoom(), False)) > 0
-                    tableWasEdited = len(this_bot.getWar().manualEdits) > 0 or len(this_bot.getRoom().dc_on_or_before) > 0 or len(this_bot.getRoom().forcedRoomSize) > 0 or this_bot.getRoom().had_positions_changed() or len(this_bot.getRoom().get_removed_races_string()) > 0 or this_bot.getRoom().had_subs() or this_bot.getRoom().race_order_changed()
+                    tableWasEdited = len(this_bot.getWar().manualEdits) > 0 or len(this_bot.getRoom().dc_on_or_before) > 0 or len(this_bot.getRoom().forcedRoomSize) > 0 or this_bot.getRoom().had_positions_changed() or len(this_bot.getRoom().get_removed_races_string()) > 0 or this_bot.getRoom().had_subs() or this_bot.getRoom().race_order_changed() or this_bot.getRoom().multiple_rxxs_used
```
Comment (BadWolf1023) [inline commands.py:1295]: See other comment. Minor change.

**Review by BadWolf1023 — CHANGES_REQUESTED** [review]
All of the old issues look good to go, however we need to change how `multiple_rxxs_used` is done as that is prone to bugs.

**Review by camelwater — APPROVED** [review]
LGTM

**Comment by BadWolf1023** [comment]
> LGTM

still some old references of `multiple_rxxs_used`, but getting close

**Comment by Fear-MK** [comment]
i did provide an update for this but idk how to merge the changes now

**Comment by Fear-MK** [comment]
> i did provide an update for this but idk how to merge the changes now

there is one line that needs to be deleted entirely, it is in the merge_room_command async function in commands.py (sets a value to True just before state is saved

one reference of multiple_rxxs_used somewhere, cant remember where

**Review by BadWolf1023 — APPROVED** [review]
Good to go imo

**Review by camelwater — APPROVED** [review]

**Pull request merged**